runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/OutputReceiverFactory.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.fnexecution.control;
+
+/**
+ * A factory that can create output receivers during an executable stage.
+ */
+public interface OutputReceiverFactory {
+  /**
+   * Get a new {@link RemoteOutputReceiver} for an output PCollection.
+   */
+  <OutputT> RemoteOutputReceiver<OutputT> create(String pCollectionId);
```

Comment:
This looks like it should be stateless, but as discussed with @bsidhom and myself, it's probable that within a single instance the caller should only ever invoke it once per `PCollection` - that ability should be commented explicitly, just so other integrators can understand permitted scoping effectively.

Reply:
fixed.